### PR TITLE
Fix implicitly bound resources to be able to reference MultiVector

### LINKED ISSUE #17: Implicit bind of vector and multivector fails
When trying to bind implicitly a vector and a multivector the generator fails. Example:
```
@bound_implicitly( characters: vertices, character_data )
archive Graph {
    vertices : vector< Character >;
    character_data: multivector< 32, Nickname, Description >;
}
```
Error:
```
12:41:29 - CRITICAL - app:67 - Error reading schema: Referring to incorrect type. Expected <class 'generator.tree.nodes.resources.vector.Vector'>, actual <class 'generator.tree.nodes.resources.multivector.Multivector'>
```

Is this expected? As far as I understand, it should be possible to bind both. In my example, I am referencing the elements of vertices and character_data with the same index.

A small sidenote: probably, the error should also refer to the line in flatdata's source, instead of the line in the source code of the generator.

### PR DESCRIPTION
**Pull request by VeaaC**
Fixes #17

### CODE REVIEW

**Comment by boxdot** [comment]
Looks good. I still would like @imagovrn to take a look.

File: tests/tree/test_syntax_tree_builder.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def test_duplicate_multivector_builtin_types_are_not_produced():
     @bound_implicitly( b: A0.v0, A0.v1 )
     archive A0 {
         v0 : vector< S1 >;
-        v1 : vector< S1 >;
+        v1 : multivector< 14, S1 >;
```
[the remark this answers is not in the document]
Comment (VeaaC) [reply tests/tree/test_syntax_tree_builder.py]: That is not a test for vector. This tree is named TREE_WITH_ALL_FEATURES

File: generator/tree/nodes/resources/bound_resource.py [hunk 2]
```diff
@@ -15,9 +15,9 @@ def create(properties, own_schema):
                              own_schema=own_schema)
 
     def _create_references(self):
-        return [VectorReference(name=r) for r in self._resources]
+        return [ResourceReference(name=r) for r in self._resources]
```
[the remark this answers is not in the document]
Comment (VeaaC) [reply generator/tree/nodes/resources/bound_resource.py]: I was thinking about that, and my reasoning was like this:
- RAW means the user manages the data
- The user might store data in RAW that use indexed
- The user might want to indicate in the schema that their RAW data is implicitly bound (however they manage their RAW data)

No strong opinions in that direction, but that seemed like a reasonable thing to allow (we do not benefit from being restrictive, do we?)

[the remark this answers is not in the document]
Comment (boxdot) [reply generator/tree/nodes/resources/bound_resource.py]: Just a small note: Right now, we already allow to create explicit references from vectors into raw data.

**Pull request merged**